src/Check.hs

```diff
@@ -90,13 +91,30 @@ checkTestsBuildReport False =
 checkTestsBuildReport True =
   "- The tests defined in `passthru.tests`, if any, passed"
 
+versionWithoutPath :: String -> Text -> String
+versionWithoutPath resultPath expectedVersion = do
+  -- We want to match expectedVersion, except when it is preceeded by
+  -- the new store path (as wrappers contain the full store path which
+  -- often includes the version)
+  -- This can be done with negative lookahead e.g
+  -- /^(?!.*${storePathWithoutVersion}).*${version}/
+  -- Note we also replace '.' with '\\.' for literal match in grep -P,
+  -- there might be a better way of escaping everything...
+  let storePath = maybe (T.pack resultPath) id $ T.stripPrefix (T.pack "/nix/store/") (T.pack resultPath)
+  let storePathWithoutVersion = maybe storePath id $ T.stripSuffix expectedVersion storePath
+  "^(?!.*"
+    <> T.unpack storePathWithoutVersion
+    <> ").*"
+    <> T.unpack (T.replace (T.pack ".") (T.pack "\\.") expectedVersion)
```

Comment:
Side note: appending (`<>`) multiple times is often less efficient than concatenating `T.concat [foo, bar, baz]`, but it would be over optimizing here. Also, for that kind of generation, one would usually leverage an interpolation library, like [PyF](https://hackage.haskell.org/package/PyF). Here `nixpkgs-update` uses in a few places [https://hackage.haskell.org/package/neat-interpolation](neat-interpolation), but it produces only `Text` and I was unable to do that (it was always complaining that some variables were not in scope, despite them actually being in scope), at least with GHC 9.0.2.